### PR TITLE
[fluxible-router] Return failure when no action can be resolved

### PR DESCRIPTION
**Pull request by tom76kimo**
@lingyan @redonkulus 

Based on #615 

Return failure when no action can be resolved

### CODE REVIEW

File: packages/fluxible-router/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "fluxible-router",
-  "version": "2.0.0-beta.3.0",
+  "version": "2.0.0-beta.4.0",
```
Comment (redonkulus) [inline packages/fluxible-router/package.json]: FYI this change will not make it into 1.x branch, which we use internally. If you need it for existing apps, you may need to backport this change to 1.x.

File: packages/fluxible-router/src/lib/navigateAction.js [hunk 2]
```diff
@@ -42,13 +42,24 @@ function navigateAction (context, payload, done) {
...
     if (!action || 'function' !== typeof action) {
-        debug('route has no action, dispatching without calling action');
-        context.dispatch('NAVIGATE_SUCCESS', completionPayload);
+        debug('action cannot be resolved');
+        completionPayload.error = {
+            statusCode: 500,
+            message: 'Action for ' + payload.url + 'can not be resolved'
+        };
+        context.dispatch('NAVIGATE_FAILURE', completionPayload);
         done();
```
Comment (lingyan) [inline packages/fluxible-router/src/lib/navigateAction.js]: we should change `done()` to pass the error

**Comment by tom76kimo** [comment]
@lingyan @redonkulus , updated! Could you help take a look again? Thank you.

File: packages/fluxible-router/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "fluxible-router",
-  "version": "2.0.0-beta.4.0",
+  "version": "1.5.4",
```
Comment (redonkulus) [inline packages/fluxible-router/package.json]: @tom76kimo  This won't work as the master branch has breaking changes for 2.x. You would need to branch from the 1.x branch and push your changes there if you want it released in the 1.x.x branch.

Comment (tom76kimo) [reply packages/fluxible-router/package.json]: Ah, you are right. Will fix it.

**Comment by pablopalacios** [comment]
I just got in the issue as well. @tom76kimo if you don't mind, I can rebase your work on top of 1.X and move this issue forward.

**Comment by tom76kimo** [comment]
@pablopalacios Please do, thank you!

**Comment by pablopalacios** [comment]
@lingyan @redonkulus it looks like we don't have a 1.x.x branch anymore, only the tag. I already rebased @tom76kimo patch on top of that tag but I cannot create a PR for it since there would be no target branch. So, would you mind to recreate the branch so we can have an 1.5.4 version released. I can also backport the change to master if it is needed but a 1.5.4 version would be awesome as well.

**Comment by lingyan** [comment]
@pablopalacios Thanks for following up on this fix!

The code change in this PR looks good, except the version update: https://github.com/yahoo/fluxible/pull/616/files#r393192433

Once that is fixed. we can merge this PR. and was talking with @redonkulus, he can create a 1.x branch for porting the fix to the 1.x version.

**Comment by redonkulus** [comment]
@pablopalacios this has been back ported and published under fluxible-router@1.5.4.

**Pull request merged**